lib/protocol/hpack/decompressor.rb

```diff
@@ -98,30 +101,56 @@ def read_header
 				pattern = peek_byte
 
 				header = {}
-				header[:type], type = HEADER_REPRESENTATION.find do |_t, desc|
-					mask = (pattern >> desc[:prefix]) << desc[:prefix]
-					mask == desc[:pattern]
-				end
 
-				raise CompressionError unless header[:type]
+				type = nil
+
+
+				case (pattern & MASK_SHIFT_4)
```

Comment:
Can you add a brief comment explaining how the lookup here works, and where the values are coming from?